designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+## Context and Scope
+
+-	https://github.com/NuGet/Home/issues/5887 
+-	https://developercommunity.visualstudio.com/content/idea/538119/add-dependency-tree-view-for-packagereferences.html 
+
+## Minimal Requirements
+
+-	Visual list of transitive dependencies.
+-	Ability to see a transitive dependency package details.
+-	Ability to determine where a transitive dependency originated.
+-	Ability to promote a transitive dependency to a top-level dependency.
+
+## Scenarios and User Experience
+
+### CLI
+
+We should consider a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency.
+
+![](../resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+### Visual Studio
+
+Within Visual Studio, there will be a new panel within the Visual Studio Package Manager UI named “Transitive Packages” in which all transitive packages will be displayed to the user. For the sake of not confusing the user, there will additionally be a header titled “Top-level Packages” where the current experience is today for top-level packages.
+
+**By default, a user’s transitive packages will be collapsed** as a project can have hundreds, if not thousands of transitive packages & would potentially impact performance displaying these many records on each load.
+
+When a user highlights a transitive package, they will see a pop-up that displays how the transitive dependency originated & what top-level package(s) are bringing it in.
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+Finally, selecting a transitive dependency will provide the user the ability to “Install” the package as you normally would a top-level dependency. This will promote the transitive dependency to a top-level dependency, and the transitive dependency will now be placed in the top-level packages list instead.
+
+## Goals and Non-Goals
+
+### Goals
+
+-	View transitive dependency list displaying the version of each dependency.
+-	See what brought in a transitive dependency & what version. 
+-	Ability to promote a transitive dependency to a top-level dependency via typical NuGet PM UI commands.
+
+### Non-Goals
+
+-	Complete management of transitive dependencies.
+-	Reinvent common UX patterns used in Visual Studio today for package management actions.
+
+## Design
+
+### Elements
+
+There are three primary elements that will be added as part of this design to the existing Visual Studio Package Manager UI.
+
+1.	ScrollViewer – To show transitive dependency nodes.
+2.	Expander – To collapse transitive dependency nodes by default & expand upon user action. To collapse top-level dependency nodes.
+3.	Tooltip – To show the transitive dependency’s origin.
```

Comment:
Mentioning the context from the other thread regarding a11y concerns, etc, would be good here.
Actually since you say below that design team has not yet decided this, we should indicate it as Optional for now?

Reply:
Thanks for the commit. This should be deferred to the implementation & the a11y concerns for whatever controls are ultimately used.